### PR TITLE
Solving #553 Addition of google oauth 2.0 for easy login 

### PR DESCRIPTION
**Pull request by migom6**
This pull request aims to solve #553 
@catarak I have implemented google oauth 2.0 using the google+ api.
please review the code and do tell me if there is need for any change.
Thanks 😄

### CODE REVIEW

**Comment by migom6** [comment]
@catarak, as for the GSOC proposal, I'll be submitting a proposal to improve the existing classrooms feature since it is intriguing, I have gone through the commits by @zrispo #425, and there is a lot to cover and improve. Since there is less time for me to code and solve the bugs, I'll focus on writing the proposal as of now.  Looking forward to contributing to this.
Thanks 😄

**Comment by shinytang6** [comment]
Hi @migom6 ! l believe some inital work has already been done by @zrispo, maybe you can try to run it locally to see the progress. Also https://medium.com/@zrispo/a-free-to-use-classrooms-system-for-p5-js-5d7b455ded08 may help :)

**Comment by migom6** [comment]
@shinytang6 Thanks for the help but i have done that already, and I have followed all his commits 😄 
Here is the brief component hierarchy.

</img>

**Comment by shinytang6** [comment]
Great! It would be better if the diagram can be more specific (just a small tip). Anyway, just waiting for @catarak to give you more suggestions :)

**Comment by migom6** [comment]
@shinytang6 Thank you for your time :)
Sorry that my diagram is not specific enough. Can you also review this PR and give more feedback?

File: README.md [hunk 1]
```diff
@@ -23,6 +23,8 @@ This project is currently in development! It will be announced when there is a (
   S3_BUCKET=<your-s3-bucket>
   GITHUB_ID=<your-github-client-id>
   GITHUB_SECRET=<your-github-client-secret>
+  GOOGLE_ID=<your-google-client-id> (use google+ api)
```
Comment (catarak) [inline README.md]: These are different from what's in `server/config/passport.js`. I think these variables should be called `GOOGLE_ID` and `GOOGLE_SECRET`, so I would change the other file.

Also, these variables need to be added to the "Production Setup" .env file example in this README.

File: client/modules/User/components/GoogleButton.jsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+// const googleUrl = require('../../../images/google.svg');
+import googlesvg from '../../../images/google.svg';
+
+function GoogleButton(props) {
+  return (
+    <a
+      className="google-button"
+      href="/auth/google/"
+    >
+      <img alt="google logo" src={googlesvg} className="google-icon" />
```
Comment (catarak) [inline client/modules/User/components/GoogleButton.jsx]: could this be an inline svg instead of an image?

**Comment by catarak** [comment]
just a few changes, but otherwise this works for me!

File: README.md (hunk 1 above)
Comment (migom6) [reply README.md]: @catarak 
Oops! my bad. I changed it. 
Thanks.

**Comment by migom6** [comment]
Hi **_@catarak_**!
I have changed the files accordingly.
Thanks for the help 😄

**Comment by migom6** [comment]
@catarak 
Just a gentle reminder. I have changed the files accordingly. 
**_cheers !_** 😄

**Comment by catarak** [comment]
this is working for me—going to merge this!

**Comment by migom6** [comment]
@catarak 
Thanks a lot for merging. **It feels awesome!**
Looking forward to more of this 😸.

**Pull request merged**